src/components/Heading/Heading.module.css

```diff
@@ -5,81 +5,64 @@
 \*------------------------------------*/
 
 /**
- * 1) h1, h2, h3, h4, h5, h6
- * 2) Styles are either controlled by as or size declaration
+ * Remove top and bottom margins applied by browser for headings.
  */
 .heading {
-  @mixin reset;
-
-  &.heading--size-1 {
-    @mixin eds-theme-typography-heading-1;
-  }
-
-  &.heading--size-2 {
-    @mixin eds-theme-typography-heading-2;
-  }
-
-  &.heading--size-3 {
-    @mixin eds-theme-typography-heading-3;
-  }
-
-  &.heading--size-4 {
-    @mixin eds-theme-typography-heading-4;
-  }
-
-  &.heading--size-5 {
-    @mixin eds-theme-typography-heading-5;
-  }
-
-  &.heading--size-6 {
-    @mixin eds-theme-typography-heading-6;
-  }
+  margin-top: 0;
+  margin-bottom: 0;
 }
 
 /**
- * H1 Heading
- * 1) Use this for the default heading if as = h1
+ * Heading sizes, correlate to similar numerically named typography token.
  */
-h1.heading {
+.heading--size-h1 {
   @mixin eds-theme-typography-heading-1;
 }
 
-/**
- * H2 Heading
- * 1) Use this for the default heading if as = h2
- */
-h2.heading {
+.heading--size-h2 {
   @mixin eds-theme-typography-heading-2;
 }
```

Comment:
This is interesting. They're ascribing styles based on the element, but then they override those with styles based on the size prop? I definitely prefer the EDS method where you define the style with one prop and then you can override the element with another prop.

Reply:
Same here